resource_elasticsearch_odfe_policy_mapping.go

```diff
@@ -0,0 +1,243 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/olivere/elastic/uritemplates"
+
+	elastic7 "github.com/olivere/elastic/v7"
+)
+
+func resourceElasticsearchOdfePolicyMapping() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceElasticsearchOdfePolicyMappingCreate,
+		Read:   resourceElasticsearchOdfePolicyMappingRead,
+		Update: resourceElasticsearchOdfePolicyMappingUpdate,
+		Delete: resourceElasticsearchOdfePolicyMappingDelete,
+		Schema: map[string]*schema.Schema{
+			"policy_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"indexes": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"include": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeMap},
+			},
+			"is_safe": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"managed_indexes": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+func resourceElasticsearchOdfePolicyMappingCreate(d *schema.ResourceData, m interface{}) error {
+	if _, err := resourceElasticsearchPostOdfePolicyMapping(d, m, "add"); err != nil {
+		if elastic7.IsNotFound(err) {
+			log.Printf("[WARN] OdfePolicyMapping (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
```

Comment:
I believe if we wanted to create an object and it wouldn't be found that we should still raise an error.

Reply:
Indeed.